lib/beaker/puppet_install_helper.rb

```diff
@@ -48,8 +48,9 @@ def run_puppet_install_helper_on(hosts, type_arg = find_install_type, version =
       if hosts_with_role(hosts, 'master').length>0 then
         # TODO Make the puppetserver code work with puppet5/puppet6
         # install puppetserver
-        install_puppetlabs_release_repo( master, 'pc1' )
+        install_puppetlabs_release_repo( master, ENV['BEAKER_PUPPET_COLLECTION'] || 'pc1')
```

Comment:
This branch of code `type == foss` appears to never be used unless you explicitly pass it as the 2nd argument to `run_puppet_install_helper_on`. Otherwise it resolves to `pe` or `agent`.

It looks like the way things actually work is that Beaker maps `BEAKER_PUPPET_COLLECTION` to `host[:puppet_collection]`, and then that's used in the helpers that install Puppet.